apps/desktop-wallet/src/api/useAddressesDisplayTokens.ts

```diff
@@ -0,0 +1,111 @@
+/*
+Copyright 2018 - 2024 The Alephium Authors
+This file is part of the alephium project.
+
+The library is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+The library is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with the library. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+import { AddressHash } from '@alephium/shared'
+import { orderBy } from 'lodash'
+import { useMemo } from 'react'
+
+import { useAddressesTokensBalances } from '@/api/addressesBalancesDataHooks'
+import { useAddressesListedFungibleTokens } from '@/api/addressesListedFungibleTokensDataHooks'
+import { useAddressesNfts } from '@/api/addressesNftsDataHooks'
+import { useAddressesTokensWorth } from '@/api/addressesTokensPricesDataHooks'
+import {
+  useAddressesUnlistedFungibleTokens,
+  useAddressesUnlistedNonStandardTokenIds
+} from '@/api/addressesUnlistedTokensHooks'
+import { ListedFTDisplay, NFTDisplay, NonStandardTokenDisplay, TokenDisplay, UnlistedFTDisplay } from '@/types/tokens'
+
+const useAddressesDisplayTokens = (addressHash?: AddressHash) => {
```

Comment:
Summarizing the talk we had on the call:

The reason of creating this hook was to get rid of the `Asset` type, which used to be a type where almost all its properties were optional to accommodate the intersection of all different type of tokens: listed FT, unlisted FT, NFTs, NSTs. Instead, the hook returns an array where each element can be any of the types I just mentioned.

The reason it is **sorted** is that we always want to display the tokens with the highest worth first, amonst other sorting criteria. That's why I went for `Display` in the name of the hook.

The reason it is a single map and not an object of maps is that in many places our current codebase relied on an array of `Asset`, so keeping it that way required the minimal amount of changes (ex: truncating the list of "assets" in address displays (address picker, address page, etc) to display on the 7 first tokens in the sorted list).

The reason for multiple hooks instead of one "massive" one is performance optimization. We don't want to unnecessarily compute data that we don't need so it's better for the component to only ask for what it only needs. 

I hope I covered everything we talked about!

Reply:
:100: 

That's what I understood, thanks for compiling it here. I'm still not convinced though :) I'd personally prefer to return a dictionary by default, and have a super simple "flatten array" hook when I need the flat list.